### PR TITLE
feat: :sparkles: Implement middleware to protect admin routes

### LINKED ISSUE #49: [FEAT] :sparkles: Implement middleware to protect admin routes
### Description

Implement a middleware that will protect routes which only admin can access

### Anything else?

get current user by using -> `res.locals.user`
and check if user role is `admin` than use `next()` function otherwise return `403` status code

### PR DESCRIPTION
**Pull request by drishit96**
### Description

Add a new middleware called '**checkAdminRole**' to check if the current user has admin access. 

### Checklist:

- [x] This pull request follow our contribution guidelines
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation (postman and swagger)
- [ ] I have written tests for the code
- [ ] I have updated .env.example file for new .env variables

### What is current behavior?

Closes #49 

### What is new behavior?

You can now protect admin only routes using the checkAdminRole middleware

### CODE REVIEW

File: src/middleware/checkAdminRole.middleware.ts [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+
+/**
+ * This middleware will check if the current user has admin role
+ */
+function checkAdminRole() {
+  return (_: Request, res: Response, next: NextFunction) => {
+    try {
+      const currentUser: User = res.locals.user;
+      if (currentUser == null || currentUser.role !== "admin") {
+        return res.sendStatus(StatusCodes.FORBIDDEN);
```
Comment (aayushchugh) [inline src/middleware/checkAdminRole.middleware.ts]: change this to

```js
return res.status(StatusCodes.FORBIDDEN).json({
error: "Insufficient rights"
})
```

File: src/middleware/checkAdminRole.middleware.ts [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+      }
+      return next();
+    } catch (err) {
+      return res.sendStatus(StatusCodes.INTERNAL_SERVER_ERROR);
```
Comment (aayushchugh) [inline src/middleware/checkAdminRole.middleware.ts]: change this to

```js
return res.status(StatusCodes.INTERNAL_SERVER_ERROR).json({
error: "Internal Server Error",
})
```

**Review by aayushchugh — CHANGES_REQUESTED** [review]

**Comment by drishit96** [comment]
@aayushchugh I don't understand why the Semantic PR CI is failing. Is there something missing in the PR?

**Comment by aayushchugh** [comment]
@drishit96 Your syntax for PR title and commits was wrong. I have updated the PR title

**Review by aayushchugh — APPROVED** [review]
@drishit96 Thanks for the PR, good job

**Comment by aayushchugh** [comment]
@all-contributors please add @drishit96 for code

**Comment by aayushchugh** [comment]
@all-contributors please add @drishit96 for code

**Comment by aayushchugh** [comment]
@all-contributors please add @drishit96 for code

**Review by BitOfAByte — APPROVED** [review]

**Comment by BitOfAByte** [comment]
Good job!

**Pull request merged**